### PR TITLE
[eas-cli] Add metadata root actions and error handling

### PR DESCRIPTION
**Pull request by byCedric**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

- Part of PR #1083
- [x] ~~Requires PR #1130~~
- [x] ~~Requires PR #1131~~
- [x] ~~Requires PR #1132~~
- [x] ~~Requires PR #1133~~
- [x] ~~Requires PR #1134~~

# How

These methods are the methods that should be used in the commands, they combine everything from the metadata into simple routines.

# Test Plan

If you go to PR #1083, you can test these actions directly from `eas metadata` and `eas metadata:configure`.

### CODE REVIEW

**Review by EvanBacon — APPROVED** [review]
add a note or open a linear task to test these after app.js, otherwise code looks solid

File: packages/eas-cli/src/metadata/download.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  );
+
+  const errors: Error[] = [];
+  const config = createAppleWriter();
+  const tasks = createAppleTasks(metadataCtx);
+  const taskCtx = { app };
+
+  for (const task of tasks) {
+    try {
+      await task.prepareAsync({ context: taskCtx });
```
[the remark this answers is not in the document]
Comment (byCedric) [reply packages/eas-cli/src/metadata/download.ts:43]: I tried to document the reason in the proposal notion, here is a summary:

We need to load certain key entities/models before we can continue to the next one. That's why I grouped that data in separate tasks. The data returned from `Task1` is then reused to fetch the data in `Task2`.

TL;DR; I agree, def would have done that if that was possible.

File: packages/eas-cli/src/metadata/download.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  for (const task of tasks) {
+    try {
+      await task.downloadAsync({ config, context: taskCtx as AppleData });
+    } catch (error: any) {
+      errors.push(error);
+    }
+  }
+
+  await fs.writeJson(filePath, config.toSchema(), { spaces: 2 });
+  unsubscribeTelemetry();
```
[the remark this answers is not in the document]
Comment (byCedric) [reply packages/eas-cli/src/metadata/download.ts:58]: Fixed in https://github.com/expo/eas-cli/pull/1147
